File: vue/src/store/entities/lesson.ts
import Entity from './entity'
import User from './user';
import Student from './student';
import Order from './order';
import Class from './class';
import TimePeriod from './timePeriod';

export default class Lesson extends Entity<number>{
    lessonDate: Date
    classId: number
    clas?: Class
    orderId: number
    order?: Order
    timePeriodId:number
    start:string
    end:string
    startTime:Date
    endTime:Date
    isFinish: boolean
    teacherId: number
    teacher?: User
    course?: string
    lastModificationTime?: string
    lastModifierUserId?: number
    creationTime?: string
    creatorUserId?: number
}

File: aspnet-core/src/EducationAdmin.Core/Education/Lesson.cs
﻿using Abp.Domain.Entities;
using Abp.Domain.Entities.Auditing;
using EducationAdmin.Authorization.Users;
using EducationAdmin.Sales;
using EducationAdmin.Settings;
using EducationAdmin.Students;
using System;
using System.Collections.Generic;
using System.Text;

namespace EducationAdmin.Education
{
    public class Lesson : FullAuditedEntity<long>, IMustHaveTenant
    {
        public int TenantId { get; set; }

        public DateTime LessonDate { get; set; }


        public long ClassId { get; set; }

        public Class Class { get; set; }

       // public int LessonNumber { get; set; }

        public bool IsFinish { get; set; }

        public long TeacherId { get; set; }

        public User Teacher { get; set; }

        public string Course { get; set; }

        public long TimePeriodId { get; set; }

        public TimePeriod TimePeriod { get; set; }

        public TimeSpan Start { get; set; }

        public TimeSpan End { get; set; }
    }
}
